wagtailsupertable/templatetags/table_span_tags.py

```diff
@@ -0,0 +1,44 @@
+import re
+from django import template
+from django.utils.safestring import mark_safe
+
+
+register = template.Library()
+
+
+@register.simple_tag(takes_context=True)
+def cell_rowspan(context, row_index, col_index, table_header=None):
+    classnames = context.get("classnames")
+    if classnames:
+        if table_header is not None:
+            row_index += 1
+        index = (row_index, col_index)
+        cell_class = classnames.get(index)
+        if cell_class and "rowspan-" in cell_class:
+            r = re.compile("rowspan-.*")
+            classes = cell_class.split()
+            rowspan = list(filter(r.match, classes))
+            if rowspan:
+                rowspan = rowspan[0].replace("rowspan-", "")
+                if rowspan != 1:
+                    return mark_safe("rowspan={}".format(rowspan))
+    return ""
+
+
+@register.simple_tag(takes_context=True)
+def cell_colspan(context, row_index, col_index, table_header=None):
+    classnames = context.get("classnames")
+    if classnames:
+        if table_header is not None:
+            row_index += 1
+        index = (row_index, col_index)
+        cell_class = classnames.get(index)
+        if cell_class and "colspan-" in cell_class:
+            r = re.compile("colspan-.*")
+            classes = cell_class.split()
+            colspan = list(filter(r.match, classes))
+            if colspan:
+                colspan = colspan[0].replace("colspan-", "")
```

Comment:
We are doing a lot of work here (splitting, checking if a substring is present, replacing text) that could also be done by the regular expression directly. This is maybe a personal preference thing, but another way of writing the regex to combine all these operations into one might be:

```python
r = re.compile(r"\bcolspan-(?P<colspan>\d+)\b")
```

I'm using a named group here and the [\b word boundary matcher](https://docs.python.org/3/howto/regex.html#more-metacharacters). This regexp lets us operate directly on the entire css class string and select part of the match we care about (the colspan amount) by name:

```python
match = r.search(cell_class)
if match:
    colspan = match.group('colspan')
```

The regular expression itself is a little more involved by in my opinion the rest of the code is cleaner. But I leave it up to you if you like this approach.

Reply:
I do like the word boundary matcher idea. Might need to just add some comments so I remember what it is doing when I come back to it.